test/sanity/validate-modules/main.py

```diff
@@ -1245,16 +1245,31 @@ def _validate_argument_spec(self, docs, spec, kwargs):
 
             # TODO: needs to recursively traverse suboptions
             doc_type = docs.get('options', {}).get(arg, {}).get('type')
-            if 'type' in data:
-                if data['type'] != doc_type and doc_type is not None:
+            if 'type' in data and data['type'] is not None:
+                if doc_type is None:
+                    if not arg.startswith('_'):  # hidden parameter, for example _raw_params
+                        self.reporter.error(
+                            path=self.object_path,
+                            code=336,
```

Comment:
Code 336 is already defined: https://docs.ansible.com/ansible/devel/dev_guide/testing_validate-modules.html

Both new codes should be added to the docs.

Reply:
Updated:
- E337/E338 used instead of E336/E337
- new codes added to the doc